src/csharp/Grpc.Core/Internal/ChannelCredentialsSafeHandle.cs

```diff
@@ -15,11 +15,16 @@
 #endregion
 using System;
 using System.Runtime.InteropServices;
-using System.Threading;
-using System.Threading.Tasks;
 
 namespace Grpc.Core.Internal
 {
+    internal delegate int VerifyPeerCallbackInternal(
```

Comment:
just use one delegate  with all the args and `isDestroy` bool flag to simplify things?

Reply:
Done. There was no `state` object like in the example you provided me, so I made use of the `userdata` pointer in the native code to keep track of the callback, I hope that's fine.